miden-lib/src/transaction/procedures.rs

```diff
@@ -0,0 +1,239 @@
+use alloc::vec::Vec;
+
+use miden_objects::{Digest, Felt, Hasher};
+
+use super::TransactionKernel;
+
+macro_rules! digest {
+    ($a:expr, $b:expr, $c:expr, $d:expr) => {
+        Digest::new([Felt::new($a), Felt::new($b), Felt::new($c), Felt::new($d)])
+    };
+}
+
+impl TransactionKernel {
+    /// Hashes of all dynamically executed procedures from the kernel 0.
+    pub const KERNEL0_PROCEDURES: [Digest; 28] = [
+        // account_vault_add_asset
+        digest!(
+            441011557193836424,
+            14128779488787237713,
+            9097945909079837843,
+            1927790173066110370
+        ),
```

Comment:
Is there a way to make it so that we can provide a single value in hex and the macro would split it? So, the input would be:
```rust
digest!(0xCAAED30CB21C7A90B979C5AC50B12C3F),
```
And the macro would translate it into:
```rust
Digest::new([
    Felt::new(0xCAAED30C),
    Felt::new(0xB21C7A90),
    Felt::new(0xB979C5AC),
    Felt::new(0x50B12C3F)
])
```

Reply:
I don't think that it is currently possible: we don't have a Felt constructor from hex, but even if we will implement it, we (almost sure) won't be able to make it constant.

But probably we can simplify the `KERNEL0_PROCEDURES` array a bit — we can store procedure hashes as felts, without collecting them into the digest, that will simplify the hash generating functions and the macro.

Reply:
> we don't have a Felt constructor from hex, but even if we will implement it, we (almost sure) won't be able to make it constant.

We don't need to create a new constructor, the current `Felt::new()` will work. It works fine now with something like `Felt::new(0x12345)`.

Reply:
Let's come back to this in a separate issue: #872